### DISCUSSION TITLE
Reborn of IC Autominer/Pump (and couple of suggestions for long distance)

### DISCUSSION THREAD

**Discussion by Pyroksyllin** [TechReborn/TechReborn#2512] [Ideas]
Can we see in the future new (or old) interpretation of IC2 autominer and/or pump? Kinda corellation with [#1683](https://github.com/TechReborn/TechReborn/issues/1683), but i don't think that filling tanks with tons of buckets manually is a good idea. By this moment i don't see any ways to have, for example, lava cells. It was good way to get energy - use lava cells, pump and geothermal generator. Autominer not so necessary as pump, in my opinion, having 3*3 mining zone for top tier drill kinda compensing this. But automatisation is good.

And long run suggestions...
Can we see sometimes some adaptations of the buildcraft features? Exactly - pipes (for fluids and items) and new types of worldspawn natural resources (oil, natural gas for example)?

btw sorry for possible lack of english in my post

**Comment by AndrewLischuk** [comment]
The question remains open. 

Although the Basic Tank pulls in fluids from other blocks (thanks to @modmuss50) . But it doesn't pulls in water (or lava) from open sources. Filling it manually is extremely uncomfortable. IMO this destroys the meaning of the automation process. An Industrial grinder, for example, requires a lot of water and now it's not possible to use it.
I agree with @Pyroksyllin  that Autominer is not so needed. But Pump is. Extremely. Or maybe it’s easier to make Basic Tank to pull in.

Pipes or tubes - awesome idea of course, but it's probably for the future. For now - Pump is needed.

For a few days I've been searching a lot of mods, but I didn't find a mod with Pump compatible for TR. Everything is either on Forge or on previous versions of MC.

Does anyone know how Drain will work?

THX)

**Comment by Tkvgamer** [comment]
simple pipes has a pump compatible with TechReborn (https://www.curseforge.com/minecraft/mc-mods/simplepipes)

**Comment by AndrewLischuk** [comment]
> simple pipes has a pump compatible with TechReborn (https://www.curseforge.com/minecraft/mc-mods/simplepipes)

Yeah, I tried this mod, but it seems doesn't work correctly on 1.15.1.... Simple Pipes's Pump filling it's tank, but ... That's it. Nothing goes on. TR's Basic Tank doesn't pulls the water from SP's Tank. Pipes doesn't work at all, or I just did something wrong... I will try ones again)))) 
Thanks for the tip, @Tkvgamer

**Comment by Sturmlilie** [comment]
@MarshallGunn you can implement a simple pump with a dispenser-tank circuit and item pipes; fits in less than half a chunk:  
![pump scheme](https://cdn.discordapp.com/attachments/159382191292416000/695716616809349200/2020-04-03_21.24.07.png)

If a dispenser contains an empty cell and gets redstone-pulsed, it will pull in the water block. Then you can send that cell to a tank and back into the dispenser (need two more pulse generators for the item pipes from SimplePipes).

About my design: There's a single 6 delay clock on the tank output pipe, the dispenser is clocked by comparator + repeater, and the dispenser output pipe is 4 ticks delayed from that signal.

**Comment by AndrewLischuk** [comment]
@Sturmlilie, hey there))) Thanks for the idea) 
I roughly understood the design, but still it is not clear to me with the red signal. Can you printscreen from different angles? And one more thing - for some reason the dispenser does not draw water into the cell or the bucket either. It simply pushes an empty object.

MC 1.15.2
TR 3.3.11 (build 224)
Simple Pipes 0.3.6

This time, I’ve better dealt with Simple Pipes. Never played with pipes before.

**Comment by Tkvgamer** [comment]
if you replace the dispenser with the pump from simplepipes and power it with a lever then you dont need the redstone circuit

**Comment by AndrewLischuk** [comment]
Guys, everything is much easier)
Thanks for helping me to figure out with Simple Pipes))))
![2020-06-10_22 39 36](https://user-images.githubusercontent.com/66631681/84311120-6d310180-ab6b-11ea-9e68-cf4e19175b4b.png)
Pump from Simple Pipes with lever and Basic Tank unit from TR (configured for fluid "auto pull in")
Now the last thing. I need to replace the textures in Simple Pipes, and it will be cool

**Comment by jusvit** [comment]
![image](https://user-images.githubusercontent.com/4002351/84560690-a9da4400-ad89-11ea-9c75-73319eadd97f.png)
****

I did experiment with a "drain" of some sorts, basically you put it under a 4x4 water source along with a tank and it'll fill it. However it's pretty useless for lava. Any ideas are welcomed.

**Comment by PhoneixS** [comment]
The creator of Simple Pipes said that they won't work on it so I think TechReborn should implement its own pipe system (and is aligned with the idea of being an independent mod).
